src/MongoDB.Driver.Core/Core/Authentication/SaslPrepHelper.cs

```diff
@@ -0,0 +1,806 @@
+﻿/* Original work:
```

Comment:
@jyemin I've only superficially reviewed this file.

But you've reviewed it carefully right?

Reply:
Yes